app/scripts/lib/xhr.js

```diff
@@ -67,6 +67,7 @@ define([
         // make sure to set the dataType for Firefox <21. See issue #1930
         dataType: 'json',
         type: options.type,
+        timeout: options.timeout || 5000,
```

Comment:
It times out at 2500 actually, this is just a default. I can reduce it further if you want.